### ISSUE TITLE
Missed events when joining a busy board

### ISSUE THREAD

**Issue by raimohanska** [raimohanska/ourboard#231]
Can reproduce when running perf-tester locally and joining a busy board.

Capture from browser console:

```
Sending board.join at serial 542
board-store.ts:373 Using local state instead of stored. Using 0 local events (out of 0)
board-store.ts:388 Going to online mode. Init at 542 with 127 new events. Board starts at 542 and first event is 543 and last 669. 0 local events queued, 0 awaiting ack.
board-reducer.ts:54 Serial skip on item.front, 669 -> 678 (firstSerial undefined serial 678)
```

**Comment by raimohanska** [comment]
After some reps with perf-tester, I've identified two variants.

Variant 1 - skip at first event received immediately after `init.diff`:

<img width="562" alt="image" src="https://github.com/raimohanska/ourboard/assets/292964/9ddb4257-3b7a-4778-8b69-7195064324bf">

Variant 2 - "negative skip" when folding events, adding to history. Seems that these are always `item.add` in my scenarios with the perf-tester.

<img width="703" alt="image" src="https://github.com/raimohanska/ourboard/assets/292964/912be259-b8b5-473a-a4d6-474824cae520">

Only variant 1 indicates missed events though, while the second variant is probably a bug in action folding. Further, the whole `addToHistory` thing is related to a feature that was removed (history view), so I will just remove the boardHistoryReducer and get better performance, less errors without any lost functionality.

**Comment by raimohanska** [comment]
There: 74693e3291fdb49aa2ce94fd323e86a17c977509. No more history tracking on the client.

**Comment by raimohanska** [comment]
In e6d75254958848903b223ed712eb17031272903e I added strict checking on the client. Now the client will re-sync in case of a serial skip, in order to recover instead of ignoring the skipped events.

The reason for the occasional missed events after `board.init.diff` is still unknown and to be investigated.

**Comment by raimohanska** [comment]
Fixed the "missed events" issue in af1a1280ffe83bbad75ef175622dd94e63ddc8ad.
